File: coco/src/main/java/coco/ep/m1/finalized/Sub.java
package coco.ep.m1.finalized;

import coco.ep.m0.finalized.Exp;

public class Sub extends Exp implements Factory, coco.ep.m1.Sub<Exp> {

    protected Exp left;

    protected Exp right;

    public Sub(Exp _left, Exp _right) {
        this.left = _left;
        this.right = _right;
    }

    public Exp getLeft() {
        return left;
    }

    public Exp getRight() {
        return right;
    }
}

File: coco_csharp/CoCo/EP/M1/Finalized/Sub.cs
namespace EP.M1.Finalized {
    public class Sub : EP.M0.Finalized.Exp, EP.M1.Sub<EP.M0.Finalized.Exp>, Factory {
        private EP.Exp<EP.M0.Finalized.Exp> left;
        private EP.Exp<EP.M0.Finalized.Exp> right;

        public Sub(EP.Exp<EP.M0.Finalized.Exp> _left, EP.Exp<EP.M0.Finalized.Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<EP.M0.Finalized.Exp> getLeft() { return this.left; }

        public EP.Exp<EP.M0.Finalized.Exp> getRight() { return this.right; }
    }
}